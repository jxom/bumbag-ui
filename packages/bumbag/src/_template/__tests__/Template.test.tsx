import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Template } from '../Template';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Template>Hello world</Template>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Template color="primary">Hello world</Template>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<Template variant="test">hello world</Template>, {
      theme: { Template: { variants: { test: { styles: { base: { backgroundColor: 'red' } } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(
      <Template overrides={{ Template: { styles: { base: { backgroundColor: 'red' } } } }}>hello world</Template>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<Template>hello world</Template>, {
      theme: { Template: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Template>hello world</Template>, {
      // @ts-ignore
      theme: { Template: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
