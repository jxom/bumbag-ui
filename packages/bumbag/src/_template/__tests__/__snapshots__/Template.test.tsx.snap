// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: var(--bb-palette-primary, #000000)!important;
}

<div
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Template"
>
  hello world
</div>
`;

exports[`overrides styles.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Template"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Template"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: var(--bb-palette-primary, #000000)!important;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Template"
>
  Hello world
</div>
`;

exports[`theming styles.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Template"
>
  hello world
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Template"
>
  hello world
</div>
`;
